### PR TITLE
Refactoring: more flexible layouts

### PR DESCRIPTION
**Pull request by angryzor**
Prepares for the addition of the second funnel, hotel booking.

This is basically finished, but I keep it as draft until the other 3 PR's are merged so it can be rebased.

### CODE REVIEW

File: src/components/funnels/funnels/register/steps/contact.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/contact.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/funnels/register/steps/optional.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/optional.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/funnels/register/steps/personal.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/personal.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/funnels/register/steps/ticket/day.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/ticket/day.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/funnels/register/steps/ticket/level.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/ticket/level.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/funnels/register/steps/ticket/level.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+const sizes = [
+	{ value: 'S', label: 'S' },
+	{ value: 'M', label: 'M' },
+	{ value: 'L', label: 'L' },
+	{ value: 'XL', label: 'XL' },
+	{ value: 'XXL', label: 'XXL' },
+]
+
+const sizesByValue = new Map(sizes.map(size => [size.value, size]))
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/ticket/level.tsx:25]: Maybe this should be moved to a constants + utils file? Seems like it's just data and a logic function

File: src/components/funnels/funnels/register/steps/ticket/type.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/funnels/register/steps/ticket/type.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/layout/common.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/layout/common.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

File: src/components/funnels/layout/with-invoice.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
+/** @jsxImportSource @emotion/react */
```
Comment (ErikMelton) [inline src/components/funnels/layout/with-invoice.tsx:1]: We shouldn't have to use `@jsxImportSource` anymore :3

**Review by ErikMelton — CHANGES_REQUESTED** [review]

File: src/components/funnels/funnels/register/steps/contact.tsx (hunk 1 above)
Comment (angryzor) [reply src/components/funnels/funnels/register/steps/contact.tsx:1]: Correct!

File: src/components/funnels/funnels/register/steps/ticket/level.tsx (hunk 6 above)
Comment (angryzor) [reply src/components/funnels/funnels/register/steps/ticket/level.tsx:25]: I could actually probably move the list of t-shirt sizes to the site config.

**Pull request merged**